### ISSUE TITLE
Identifo Web Login: 500 error when trying to login with email and password

### ISSUE THREAD

**Issue by kazemisoroush** [MadAppGang/identifo#187]
When I try to login with Email & Password pair, I'm seeing a 500 error in API response:

```
PANIC: runtime error: invalid memory address or nil pointer dereference
goroutine 940 [running]:
github.com/urfave/negroni.(*Recovery).ServeHTTP.func1(0x7fb45fefd4b8, 0xc000218068, 0xc0001fc1e0, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/recovery.go:159 +0xd2
panic(0x10355c0, 0x1a725e0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/casbin/casbin/persist.LoadPolicyLine(0xc0005c61e4, 0x15, 0xc000289950)
	/go/pkg/mod/github.com/casbin/casbin@v1.9.1/persist/adapter.go:36 +0x223
github.com/qiangmzsx/string-adapter.(*Adapter).LoadPolicy(0xc000142e20, 0xc000289950, 0x109faa0, 0x1099b00)
	/go/pkg/mod/github.com/qiangmzsx/string-adapter@v0.0.0-20180323073508-38f25303bb0c/adapter.go:48 +0xd3
github.com/casbin/casbin.(*Enforcer).LoadPolicy(0xc000267180, 0x7fb45ff03658, 0xc000142e20)
	/go/pkg/mod/github.com/casbin/casbin@v1.9.1/enforcer.go:214 +0x53
github.com/casbin/casbin.(*Enforcer).InitWithModelAndAdapter(0xc000267180, 0xc000289950, 0x7fb45ff03658, 0xc000142e20)
	/go/pkg/mod/github.com/casbin/casbin@v1.9.1/enforcer.go:128 +0x2d7
github.com/casbin/casbin.NewEnforcer(0xc00089af10, 0x2, 0x2, 0xc000289950)
	/go/pkg/mod/github.com/casbin/casbin@v1.9.1/enforcer.go:81 +0x2cb
github.com/madappgang/identifo/web/authorization.(*Authorizer).initInternalAuthorizer(0xc000674020, 0xc000834978, 0x18, 0xc000834990, 0x18, 0x1, 0xc000839b34, 0x5, 0xc0000a8b40, 0x2d, ...)
	/go/src/github.com/madappgang/identifo/web/authorization/authorization.go:147 +0x258
github.com/madappgang/identifo/web/authorization.(*Authorizer).authorizeInternal(0xc000674020, 0xc000834978, 0x18, 0xc000834990, 0x18, 0x1, 0xc000839b34, 0x5, 0xc0000a8b40, 0x2d, ...)
	/go/src/github.com/madappgang/identifo/web/authorization/authorization.go:114 +0x65
github.com/madappgang/identifo/web/authorization.(*Authorizer).Authorize(0xc000674020, 0xc000834978, 0x18, 0xc000834990, 0x18, 0x1, 0xc000839b34, 0x5, 0xc0000a8b40, 0x2d, ...)
	/go/src/github.com/madappgang/identifo/web/authorization/authorization.go:46 +0xfc
github.com/madappgang/identifo/web/api.(*Router).LoginWithPassword.func1(0x7fb45fefd4b8, 0xc000218068, 0xc00020bb00)
	/go/src/github.com/madappgang/identifo/web/api/login.go:150 +0x4d8
net/http.HandlerFunc.ServeHTTP(0xc000347b50, 0x7fb45fefd4b8, 0xc000218068, 0xc00020bb00)
	/usr/local/go/src/net/http/server.go:2049 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001e6600, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b900)
	/go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xd3
github.com/urfave/negroni.Wrap.func1(0x7fb45fefd4b8, 0xc000218068, 0xc00020b900, 0xc000599400)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:46 +0x4d
github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc00000e990, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b900, 0xc000599400)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:29 +0x4e
github.com/urfave/negroni.middleware.ServeHTTP(0x14145e0, 0xc00000e990, 0xc00000ea98, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b900)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
github.com/madappgang/identifo/web/api.(*Router).SignatureHandler.func1(0x7fb45fefd4b8, 0xc000218068, 0xc00020b900, 0xc0005993e0)
	/go/src/github.com/madappgang/identifo/web/api/appsecret.go:80 +0x6ad
github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc000347b00, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b900, 0xc0005993e0)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:29 +0x4e
github.com/urfave/negroni.middleware.ServeHTTP(0x14145e0, 0xc000347b00, 0xc00000ea80, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b900)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
net/http.HandlerFunc.ServeHTTP(...)
	/usr/local/go/src/net/http/server.go:2049
github.com/madappgang/identifo/web/api.(*Router).AppID.func1(0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599380)
	/go/src/github.com/madappgang/identifo/web/api/app_middleware.go:35 +0x32d
github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc0003479e0, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599380)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:29 +0x4e
github.com/urfave/negroni.middleware.ServeHTTP(0x14145e0, 0xc0003479e0, 0xc00000ea50, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
github.com/rs/cors.(*Cors).ServeHTTP(0xc000666000, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599360)
	/go/pkg/mod/github.com/rs/cors@v1.6.0/cors.go:240 +0x18f
github.com/urfave/negroni.middleware.ServeHTTP(0x1412c20, 0xc000666000, 0xc00000ea38, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
net/http.HandlerFunc.ServeHTTP(...)
	/usr/local/go/src/net/http/server.go:2049
github.com/madappgang/identifo/web/api.(*Router).RemoveTrailingSlash.func1(0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599340)
	/go/src/github.com/madappgang/identifo/web/api/app_middleware.go:41 +0x82
github.com/urfave/negroni.HandlerFunc.ServeHTTP(0x11d5940, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599340)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:29 +0x4e
github.com/urfave/negroni.middleware.ServeHTTP(0x14145e0, 0x11d5940, 0xc00000ea08, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
github.com/urfave/negroni.(*Static).ServeHTTP(0xc0006eacc0, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599320)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/static.go:34 +0x7c
github.com/urfave/negroni.middleware.ServeHTTP(0x1412cc0, 0xc0006eacc0, 0xc00000e9f0, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
github.com/urfave/negroni.(*Logger).ServeHTTP(0xc0006ea720, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc000599300)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/logger.go:62 +0x8a
github.com/urfave/negroni.middleware.ServeHTTP(0x1412c60, 0xc0006ea720, 0xc00000e9c0, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
github.com/urfave/negroni.(*Recovery).ServeHTTP(0xc0001fc1e0, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800, 0xc0005992e0)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/recovery.go:193 +0x8c
github.com/urfave/negroni.middleware.ServeHTTP(0x1412ca0, 0xc0001fc1e0, 0xc00000e9a8, 0x7fb45fefd4b8, 0xc000218068, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:38 +0x9c
github.com/urfave/negroni.(*Negroni).ServeHTTP(0xc0006eb080, 0x1423830, 0xc000018460, 0xc00020b800)
	/go/pkg/mod/github.com/urfave/negroni@v1.0.0/negroni.go:96 +0xf1
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001e6540, 0x1423830, 0xc000018460, 0xc00020b600)
	/go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212 +0xd3
github.com/madappgang/identifo/web/api.(*Router).ServeHTTP(0xc00079c000, 0x1423830, 0xc000018460, 0xc00020b600)
	/go/src/github.com/madappgang/identifo/web/api/router.go:38 +0x4d
net/http.(*ServeMux).ServeHTTP(0xc0005ea1c0, 0x1423830, 0xc000018460, 0xc00020b600)
	/usr/local/go/src/net/http/server.go:2428 +0x1ad
github.com/madappgang/identifo/web.(*Router).ServeHTTP(0xc00043def0, 0x1423830, 0xc000018460, 0xc00020b600)
	/go/src/github.com/madappgang/identifo/web/router.go:95 +0x4d
net/http.serverHandler.ServeHTTP(0xc0004ec1c0, 0x1423830, 0xc000018460, 0xc00020b600)
	/usr/local/go/src/net/http/server.go:2867 +0xa3
net/http.(*conn).serve(0xc0002db860, 0x1428620, 0xc000331000)
	/usr/local/go/src/net/http/server.go:1932 +0x8cd
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2993 +0x39b
```

Here is a breakdown on request and headers:
![image](https://user-images.githubusercontent.com/8931595/130164593-e02c5996-ee76-4371-8040-5437820200cd.png)

Demo in this video:
https://www.loom.com/share/df2886204d8941539b6a795b0792216b

**Comment by kazemisoroush** [comment]
Also, there is no way to login with username or mobile number on web:
![image](https://user-images.githubusercontent.com/8931595/130164752-d8bc930b-8727-4e0a-9640-5ebe019c6fc1.png)
![image](https://user-images.githubusercontent.com/8931595/130164780-ebb07b3d-fe20-4fc7-a82c-212f5be6942b.png)

**Comment by sokolovstas** [comment]
This is casbin error. Maybe this bug is connected to #183 after failed save request. Please check that login is working when authorization is disabled or on fresh created application.

Second part will be fixed in #180

**Comment by sokolovstas** [comment]
I will close this issue for now, if error persist feel free to open
